src/sets.jl

```diff
@@ -251,6 +301,14 @@ end
 dual_set(s::SecondOrderCone) = copy(s)
 dual_set_type(::Type{SecondOrderCone}) = SecondOrderCone
 
+function distance_to_set(v::AbstractVector{<:Real}, s::SecondOrderCone)
+    _check_dimension(v, s)
+    t = v[1]
+    xs = v[2:end]
+    result = LinearAlgebra.norm(xs) - t
+    return max(result, zero(result)) # avoids sqrt
```

Comment:
Why return a vector of distances for `Nonnegatives` and a scalar for `SecondOrderCone`? Both are convex cones.

Reply:
SOC yields one result (there is one inequality to respect to belong to the set), versus Nonnegative where `dimension` inequalities must be respected